tensorflow_addons/optimizers/average_wrapper.py

```diff
@@ -103,14 +104,17 @@ def assign_average_vars(self, var_list):
         model.save('model.h5')
         ```
         """
-        assign_op = tf.group(
-            [
-                var.assign(self.get_slot(var, "average"), use_locking=self._use_locking)
-                for var in var_list
-                if var.trainable
-            ]
-        )
-        return assign_op
+        assign_ops = []
+        for var in var_list:
+            try:
+                assign_ops.append(
+                    var.assign(
+                        self.get_slot(var, "average"), use_locking=self._use_locking
+                    )
+                )
+            except Exception as e:
+                warnings.warn("Unable to assign average slot to {} : {}".format(var, e))
```

Comment:
It's better to use ```tf.get_logger().warning```?

Reply:
Seems that other than `seq2seq`, most components in addons use `warnings.warn` print warnings. We should make it consistent.